components/automate-cli/pkg/verifyserver/services/hardwareresourcecount/hardwareresourcecount.go

```diff
@@ -0,0 +1,201 @@
+package hardwareresourcecount
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/lib/logger"
+)
+
+type IHardwareResourceCountService interface {
+	GetHardwareResourceCount(models.HardwareResourceRequest) []models.HardwareResourceResponse
+}
+
+type HardwareResourceCountService struct {
+	log logger.Logger
+}
+
+func NewHardwareResourceCountService(log logger.Logger) *HardwareResourceCountService {
+	return &HardwareResourceCountService{
+		log: log,
+	}
+}
+
+func RunHardwareResourceCountCheck(minNodeCount, reqNodeCount int, nodetype string, ip string, set, setOfCluster map[string]string, ch chan map[string]models.HardwareResourceResponse, key string) {
+	response := ValidateHardwareResources(minNodeCount, reqNodeCount, nodetype, ip, set, setOfCluster)
+	respMap := make(map[string]models.HardwareResourceResponse)
+	respMap[key] = response
+	ch <- respMap
+}
+
+// This function is mainly used for calling the 4 main checks, and preparing the response.
+func ValidateHardwareResources(minNodeCount, reqNodeCount int, nodetype string, ip string, set, setOfCluster map[string]string) models.HardwareResourceResponse {
+	var res = models.HardwareResourceResponse{}
+	res.NodeType = nodetype
+	res.IP = ip
+
+	checks := UniqueIP(nodetype, len(set), reqNodeCount)
+	res.Checks = append(res.Checks, checks)
+
+	checks = ValidFormat(ip)
+	res.Checks = append(res.Checks, checks)
+
+	checks = SharedIP(nodetype, ip, setOfCluster)
+	res.Checks = append(res.Checks, checks)
+
+	checks = ValidCount(minNodeCount, reqNodeCount, nodetype)
+	res.Checks = append(res.Checks, checks)
+
+	return res
+}
+
+// This function will check that are we getting unique ips for each service. Like for automate if we get node count 2 in request, then we are also getting 2 unique ips of automate.
+func UniqueIP(nodetype string, set int, nodecount int) models.Checks {
+	var check = models.Checks{}
+	if set == nodecount {
+		check = createCheck(constants.IP_ADDRESSS, true, constants.UNIQUE_SUCCESS_MESSAGE, "", "")
+	} else {
+		check = createCheck(constants.IP_ADDRESSS, false, "", constants.UNIQUE_ERROR_MESSAGE, fmt.Sprintf(constants.UNIQUE_RESOLUTION_MESSAGE, nodetype))
+	}
+	return check
+}
+
+func ValidFormat(ip string) models.Checks {
+	var check = models.Checks{}
+	if net.ParseIP(ip) == nil {
```

Comment:
instead of this if-else block, you can write as below
`if net.ParseIP(ip) == nil {
		return createCheck(constants.IP_ADDRESSS, false, "", constants.VALID_FORMAT_ERROR_MESSAGE, fmt.Sprintf(constants.VALID_FORMAT_RESOLUTION_MESSAGE, ip))
	} 
	return  createCheck(constants.IP_ADDRESSS, true, constants.VALID_FORMAT_SUCCESS_MESSAGE, "", "")
`
same for other checks as well